src/lib/pages/backoffice/Actions/CheckIn/ItemsSearch/state/actions.js

```diff
@@ -0,0 +1,174 @@
+import { itemApi } from '@api/items';
+import { loanApi } from '@api/loans';
+import { documentApi } from '@api/documents';
+import {
+  sendErrorNotification,
+  sendSuccessNotification,
+  addNotification,
+} from '@components/Notifications';
+import { toShortDate } from '@api/date';
+
+export const IS_LOADING = 'itemsSearch/IS_LOADING';
+export const SUCCESS = 'itemsSearch/SUCCESS';
+export const HAS_ERROR = 'itemsSearch/HAS_ERROR';
+export const QUERY_STRING_UPDATE = 'itemsSearch/QUERY_STRING_UPDATE';
+export const LOAN_IS_LOADING = 'itemLoansSearch/IS_LOADING';
+export const LOAN_SUCCESS = 'itemLoansSearch/SUCCESS';
+export const LOAN_HAS_ERROR = 'itemLoansSearch/HAS_ERROR';
+export const MULTIPLE_LOAN_RESULTS = 'itemLoansSearch/MULTIPLE_LOAN_RESULTS';
+export const CHECKIN_IS_LOADING = 'itemCheckin/IS_LOADING';
+export const CHECKIN_SUCCESS = 'itemCheckin/SUCCESS';
+export const CHECKIN_HAS_ERROR = 'itemCheckin/HAS_ERROR';
+export const CLEAR = 'itemCheckin/CLEAR';
+export const CLEAR_SEARCH = 'itemCheckin/CLEAR_SEARCH';
+
+export const checkin = barcode => {
+  return async dispatch => {
+    dispatch({
+      type: IS_LOADING,
+    });
+
+    try {
+      const response = await itemApi.list(
+        itemApi
+          .query()
+          .withBarcode(barcode)
+          .qs()
+      );
+
+      dispatch(fetchItemLoans(response.data.hits[0].metadata.pid));
+
+      dispatch({
+        type: SUCCESS,
+        payload: response.data,
+      });
+    } catch (error) {
+      if (!error.response) {
+        dispatch(
+          addNotification('Error', `Barcode ${barcode} doesn't exist.`, 'error')
+        );
+      }
+      dispatch({
+        type: HAS_ERROR,
+        payload: error,
+      });
+      dispatch(sendErrorNotification(error));
+    }
+  };
+};
+
+export const updateQueryString = queryString => {
+  return dispatch => {
+    dispatch({
+      type: QUERY_STRING_UPDATE,
+      queryString: queryString,
+    });
+  };
+};
+
+export const fetchItemLoans = itemPid => {
+  return async dispatch => {
+    dispatch({
+      type: LOAN_IS_LOADING,
+    });
+
+    try {
+      let response = await loanApi.list(
+        loanApi
+          .query()
+          .withItemPid(itemPid)
+          .withState('ITEM_ON_LOAN')
+          .qs()
+      );
+
+      if (response.data.hits.length > 1) {
+        response.data.hits.forEach(item => {
+          item.metadata.start_date = toShortDate(item.metadata.start_date);
+          item.metadata.end_date = toShortDate(item.metadata.end_date);
+        });
+
+        dispatch({
+          type: MULTIPLE_LOAN_RESULTS,
+          payload: response.data,
+        });
+      } else {
```

Comment:
if there are multiple loans I display a list, if there is one loan I check it in and if there are zero it will try to read the metadata here  
https://github.com/inveniosoftware/react-invenio-app-ils/blob/d0288391822db99f2c8e3a7bde8b201aae493ab7/src/lib/pages/backoffice/Actions/CheckIn/ItemsSearch/state/actions.js#L95
and throw and error.